sdk/textanalytics/Azure.AI.TextAnalytics/tests/samples/SampleMoq.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Testing;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.AI.TextAnalytics.Samples
+{
+    [LiveOnly]
+    public class SampleMoq
+    {
+        [Test]
+        public async Task DetectLanguageAsync()
+        {
+            #region Snippet:CreateMocks
+            var mockResponse = new Mock<Response>();
+            var mockClient = new Mock<TextAnalyticsClient>();
+            #endregion
+
+            #region Snippet:SetupMocks
+            Response<DetectedLanguage> response = Response.FromValue(TextAnalyticsModelFactory.DetectedLanguage("Spanish", "es", 1.00), mockResponse.Object);
+
+            mockClient.Setup(c => c.DetectLanguageAsync("Este documento está en español.", It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                .Returns(Task.FromResult(response));
+            #endregion
+
+            #region Snippet:UseMocks
+            TextAnalyticsClient client = mockClient.Object;
+            bool result = await IsSpanishAsync("Este documento está en español.", client, default);
+            Assert.IsTrue(result);
+            #endregion
+        }
+
+        #region Snippet:MethodToTest
+        private static async Task<bool> IsSpanishAsync(string document, TextAnalyticsClient client, CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                DetectedLanguage language = await client.DetectLanguageAsync(document);
+                return language.Iso6391Name == "es";
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            return false;
+        }
+        #endregion Snippet:MethodToTest
+
+        [Test]
+        public async Task DetectLanguageBatchAsync()
```

Comment:
Should we produce any documentation for this test?

Reply:
The intention is just to show what other types of things to do. But not necessarily put them in a snippet.

Reply:
Ok